src/DynamoCoreWpf/UI/HeaderTemplateSelector.cs

```diff
@@ -0,0 +1,25 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Windows;
+using System.Windows.Controls;
+using Dynamo.UI.Controls;
+
+namespace Dynamo.Controls
+{
+    public class HeaderTemplateSelector : DataTemplateSelector
```

Comment:
Thanks @vmoyseenko, I have discussed this with @elayabharath. Based on my understanding here are the differences between *single-item list* and *multiple-item list*:

1. Items will have border when there are multiple items.
2. Items will be clickable in multiple item list.
3. Items will have hover-over effects in multiple item list.

The above can be achieved in the following way, respectively:

1. Border can be styled based on item count (more than 1 item, show the border).
2. Click events can be ignored (or not sent) if there is only one item.
3. Hover color can be made the same as normal color if there is only 1 item in the list.

Does that make sense? It will simplify your codes further. Please let me know what you think.